### PR TITLE
feat: use pgx db driver

### PR DESCRIPTION
**Pull request by commoddity**
This PR adds the updated PGX Portal DB driver.

Along with the update I've set up a configurable `APP_ENV` env var which determines the DB connection method to use, which is as follows:

- APP_ENV = `test` or `development` - Will connect to the database using standard Postgres Username/PW connection. Currently should just be used to connect to the local Docker DB for running the E2E tests.
- APP_ENV = `production` - Will connect using the Google CloudSQL connector lib, which is what we will use in production.

### CODE REVIEW

File: cache/cache.go [hunk 1]
```diff
@@ -870,6 +870,9 @@ func (c *Cache) addChainGigastakeApp(chainGigastake types.ChainGigastakeApp) {
 		return
 	}
 
+	if gigastakeApp.ChainIDs == nil {
```
Comment (rem1niscence) [inline cache/cache.go:873]: Is this strongly related to the PR or a quick fix?

Comment (commoddity) [reply cache/cache.go:873]: Just caused one of the E2E tests to fail so I added the check.

**Review by rem1niscence — APPROVED** [review]
